00_Base/src/util/AsyncResponder.ts

```diff
@@ -5,70 +5,35 @@ import { ActiveChargingProfileResult } from '../model/ActiveChargingProfileResul
 import { ClearChargingProfileResult } from '../model/ChargingprofilesClearProfileResult';
 import { ChargingProfileResult } from '../model/ChargingProfileResult';
 import { NotFoundError } from 'routing-controllers';
-import { OcpiParams } from '../trigger/util/ocpi.params';
 import { CommandResult } from '../model/CommandResult';
-import { SessionMapper } from '../mapper/session.mapper';
-import { SequelizeTransactionEventRepository } from '@citrineos/data';
+import { OcpiParams } from '../trigger/util/ocpi.params';
 
 @Service()
 export class AsyncResponder {
   constructor(
     readonly responseUrlRepo: ResponseUrlRepository,
     readonly asyncResponseApi: AsyncReceiverApi,
-    readonly sessionMapper: SessionMapper,
-    readonly transactionRepo: SequelizeTransactionEventRepository,
   ) {}
 
-  async send(correlationId: string, data: CommandResult) {
-    const responseUrlEntity =
-      await this.responseUrlRepo.getResponseUrl(correlationId);
-    if (responseUrlEntity) {
-      await this.asyncResponseApi.postAsyncResponse(
-        responseUrlEntity.responseUrl,
-        data,
-      );
-    } else {
-      throw new NotFoundError(
-        'No response url found for correlationId: ' + correlationId,
-      );
-    }
-  }
-
-  async sendChargingProfileResult(
+  async send(
     correlationId: string,
     data:
       | ActiveChargingProfileResult
       | ClearChargingProfileResult
-      | ChargingProfileResult,
+      | ChargingProfileResult
+      | CommandResult,
+    params?: OcpiParams,
   ) {
     const responseUrlEntity =
       await this.responseUrlRepo.getResponseUrl(correlationId);
     if (responseUrlEntity) {
-      if (!responseUrlEntity.sessionId) {
-        throw new NotFoundError('Session Id not found');
+      params = params ?? responseUrlEntity.params;
```

Comment:
Minor, but I wonder if it is cleaner to always save the OcpiParams to response url entity.

Then, you would never need to pass into this method.

This would require you to always set the ocpi params from commands executer.

Reply:
> Minor, but I wonder if it is cleaner to always save the OcpiParams to response url entity.
> 
> Then, you would never need to pass into this method.
> 
> This would require you to always set the ocpi params from commands executer.

This param is set as an optional column in table because in use case like start session command, the transaction id is not yet generated when we store the response url.

Reply:
Yes transaction is not yet generated, but we would still know the ocpi params as this is given in the start session url params.

We just dont pass it down from the api layer.

I am not sure if @ChrisWeissmann or @elliot-sabitov implemented something where we can easily extract the ocpi headers from anywhere in the code, or if we'd be required to pass it down.

That being said, just a minor thought I had. PR looks good.

Reply:
yep, if this is a Koa endpoint, you should be able to get it like this: ```@FunctionalEndpointParams() ocpiHeaders?: OcpiHeaders,``` in the request handler method

Reply:
If we change to use `ocpiHeaders`, we would need to pass headers from request handler down to service and then command executor. I think we can keep the existing change for now and decide whether refactor to use `ocpiHeaders` for Command and Charging Profile apis later.

Reply:
I agree @lydiazcheng